### ISSUE TITLE
Bug report: unable to catch / ignore failed requests resulting in a non-200 HTTP code

### ISSUE THREAD

**Issue by Pampattitude** [wkhtmltopdf/wkhtmltopdf#1877]
Hi there!

I have two bugs to report:
<code>wkhtmltoimage --format jpeg --load-media-error-handling ignore --load-error-handling ignore test.html test.jpg</code>
... with the following HTML file:

```html
<html>
  <head>
    <script type="application/javascript" src="http://atout.email-match.com/"></script>
  </head>
  <body>
  </body>
</html>
```

... results in the following:

<pre><code>Loading page (1/2)
Rendering (2/2)
Done
Exit with code 1 due to network error: ContentOperationNotPermittedError
</code></pre>

Note that the test page (<code>atout.email-match.com</code>) returns a custom 403 page with a 403 HTTP error code.

Desired result would either be that the error is ignored, thanks to <code>--load-error-handling</code>, or another command line option could be used to define how to handle such errors.

Now, for the second bug, same situation but with <code>file:</code> as the protocol for the JavaScript script instead of <code>http:</code>.
Results in the following:

<pre><code>Loading page (1/2)
Rendering (2/2)
Done
Exit with code 1 due to network error: ProtocolInvalidOperationError
</code></pre>

Still no way to define how to handle this kind of errors.

And, finally, I don't get the difference between <code>ignore</code> and <code>skip</code> values for <code>*-error-handling</code> options, and could not find anything about it in the manual. What's the difference?

Thank you very much for your time (and for making wkhtmlto*)! :3

[EDIT] Forgot the wkhtmltoimage version, here it goes:
<code>wkhtmltoimage 0.12.2-6a13a51 (with patched qt)</code>
<code>Mac OS X v10.7.5 build 11G63b</code>

**Comment by mwtorkowski** [comment]
+1 to that - I'm experiencing the same issue, version 0.12.1 with patched qt.
Image url returns custom 403 page (with 403 http response code), this flips off wkhtmltopdf.

**Comment by ashkulz** [comment]
I'm pretty sure that this happens when there is no extension in the URL to help determine whether it is a media request or not.

**Comment by mwtorkowski** [comment]
Interesting observation, I have to test this tomorrow. Thanks!
8 sty 2015 20:27 "Ashish Kulkarni" <notifications@github.com> napisał(a):

**Comment by mwtorkowski** [comment]
I've checked what @ashkulz suggested and it doesn't seem to be the case. What's more, I searched wkhtmltopdf repo as welll as qt repository for references to --load-media-error-handling option (I tried variations of this string), and there's no mention of that option. That, plus the results I've seen when searching for ContentOperationNotPermittedError in the repos, leads me to believing that this option is a dead, obsolete feature.
As can be seen here https://github.com/wkhtmltopdf/qt/search?utf8=%E2%9C%93&q=ContentOperationNotPermittedError this error is returned for various cases, 403 error among them.

**Comment by vigintas** [comment]
Firs of all - thank you for the amazing work and tool that I use every day!

But I am too having ContentNotFoundError on few of the conversions. Upon inspection the html had an image that returned 403. I'm not in control of the server that serves images.

Tried --load-error-handling ignore/skip and --load-media-error-handling ignore/skip with no success, seems these flags are just ignored. The image link has a *.png extension so it should be perceived as media I'd assume.

As a workaround I had to manually check the image response and replace the url with something default if 403 was returned. As you can imagine this looks and feels really ugly. Besides it works only if the exact same image is in the exact same place in order for me to be able to check the response.

Any update or a cleaner workaround would be highly appreciated!

wkhtmltopdf 0.12.1 (with patched qt)

**Comment by omiras** [comment]
If it helps someone:
I had a similar issue.

Execution of wkhtmltopdf in command line did throw a ProtocolInvalidOperationError error, but also generated the PDF file.  This is my command line:

wkhtmltopdf --orientation Portrait --page-size A4 --zoom 1 --quiet  --title "CV: 1262D35810994F98_2698640" file.html file.pdf

By the way, I also wanted to use Wkhtmltopdf from PHP. I had to use XVFB  to get rid off ProtocolInvalidOperationError:

$pdf = new Wkhtmltopdf(
        array(
        'path' => sys_get_temp_dir(),
        'xvfb' => TRUE,
        )
      );

**Comment by OpenESignForms** [comment]
We see this issue still on wkhtmltopdf 0.12.3 (with patched qt)

For us, the issue is the HTML points to an image file that is no longer in the location where it was when the HTML was first retrieved.  We noted that the "--load-error-handling ignore --load-media-error-handling ignore" seems to have no effect (same with 'skip' which would seem very much the same meaning as ignore).

Interestingly, a PDF is generated even with the exit status 1, but currently our code is only assuming a valid PDF when exit status is 0.  We are reluctant to give a user a PDF after an exit status of 1 now because it was our thought that it mean the conversion failed, not just had a warning type error.

Does exit status 1 mean there were these sorts of error but a PDF was generated?  If there a different code for when the PDF generation itself fails?

**Comment by alloylab** [comment]
are you still have this issues in 0.12.5?

**Comment by kennedy-osaze** [comment]
yes i am @alloylab

**Comment by iJanki** [comment]
I still have the same issues using master branch.

**Comment by jonathanbull** [comment]
Same issue here. Any update on this?

**Comment by jasperpg** [comment]
Yes, please anyone can define what's the difference between 'skip' and 'ignore' for both load-media-error-handling and load-error-handling?

**Comment by bobintetley** [comment]
I'm having to parse documents and test/remove dead image links before feeding them to wkhtmltopdf because ignore/skip does not work. I have the issue with 0.12.3, 0.12.4, 0.12.5 and master.
